src/var/mutation_counting.jl

```diff
@@ -0,0 +1,152 @@
+# phylo/mutation_counting.jl
+# ==================
+#
+# Types and methods for counting different kinds of mutations between two
+# DNA or RNA sequences.
+#
+# Part of the Bio.Phylo module.
+#
+# This file is a part of BioJulia. License is MIT: https://github.com/BioJulia/Bio.jl/blob/master/LICENSE.md
+
+
+# Counting mutations
+# ------------------
+
+# Mutation types
+
+immutable Different end
+immutable Transition end
+immutable Transversion end
```

Comment:
These names are too generic. Names like `TransitionMutation` would be better I think.